JICMP-26: Antora (asciidoc) documentation, plus CI updates (#5)
OpenNMS/jicmp: docs-src/modules/using/pages/non-root.adoc

= Using JICMP as non-root

JICMP normally uses raw sockets, which requires root access on some operating systems.

== Mac OS X

> _Mac OS X_ supports non-root _ICMP_ through the _$$SOCK_DGRAM$$_ interface, which _JICMP_ uses by default if it is found.

Reviewer: ```suggestion
Mac OS X supports non-root ICMP through the $$SOCK_DGRAM$$ interface, which JICMP uses by default if it is found.
```

== Linux

On Linux, you must set a `sysctl` `OID` to allow ping from non-root users.

> To do so, you specify a range of `gid` values that are allowed to send non-root. The examples below use `429496729` because it is the highest possible `gid` value, but it is instead recommended that you specify the `gid` of the user OpenNMS will run as instead. This limits the number of accounts allowed to access raw sockets on the system.

Reviewer: ```suggestion
To do so, specify a range of `gid` values that are allowed to send non-root. 
The examples below use `429496729` because it is the highest possible `gid` value, but we recommend that you specify the `gid` of the user that OpenNMS will run as instead.
This limits the number of accounts allowed to access raw sockets on the system.
```

> This can be accomplished one-time by running:

Reviewer: ```suggestion
To do so, run the following:
```
